src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1229,12 +1254,32 @@ private void CheckRestrictedTypeReceiver(BoundExpression expression, CSharpCompi
                     GetOriginalMethods(result), methodGroup.ResultKind, methodGroup.TypeArguments.ToImmutable(), analyzedArguments, invokedAsExtensionMethod: invokedAsExtensionMethod, isDelegate: ((object)delegateTypeOpt != null));
             }
 
-            // Otherwise, there were no dynamic arguments and overload resolution found a unique best candidate. 
+            // Otherwise, overload resolution found a unique best candidate. 
             // We still have to determine if it passes final validation.
 
             var methodResult = result.ValidResult;
             var returnType = methodResult.Member.ReturnType;
             var method = methodResult.Member;
+            bool forceDynamicResultType = false;
+
+            var useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+
+            // Due to backward compatibility, invocations statically bound in presence of dynamic arguments
+            // should have dynamic result if their dynamic binding succeeded in C# 12 and there are no
+            // obvious reasons for the runtime binder to fail (ref return, for example).
+            if (hasDynamicArgument &&
+                !(methodGroup.IsExtensionMethodGroup || method.MethodKind == MethodKind.LocalFunction ||
```

Comment:
>methodGroup.IsExtensionMethodGroup [](http://example.com/codeflow?start=18&length=34)

Shouldn't extension methods get dynamified?

Reply:
From offline discussion, this falls out of proposed spec. We'll leave question of whether the result should be converted to dynamic in C# 13 and forward to LDM discussion.